src/main/thrift/parquet.thrift

```diff
@@ -595,6 +607,43 @@ union ColumnOrder {
   1: TypeDefinedOrder TYPE_ORDER;
 }
 
+struct PageLocation {
+/** Offset of the page in the file **/
+  1: required i64 offset
+
+/** Size of the page, including header. The same as PageHeader.compressed_page_size **/
+  2: required i32 compressed_page_size
+
+/** Index within the RowGroup of the first row of the page **/
+  3: required i64 first_row_index
+}
+
+struct OffsetIndex {
+/** PageLocations, ordered by increasing PageLocation.offset **/
+  1: required list<PageLocation> page_locations
+}
+
+/**
+ * Description for ColumnIndex.
+ * Each <array-field>[i] refers to the page at OffsetIndex.page_locations[i]
+ */
+struct ColumnIndex {
+/** A list of bools to determine the validity of the corresponding min and max values **/
+  1: required list<bool> valid_values
+
+/** A list containing the lower bounds for the values of each page **/
+  2: required list<binary> min_values
+
+/** A list containing the upper bounds for the values of each page **/
+  3: optional list<binary> max_values
+
+/** A list containing DataPageHeaderV2.statistics.null_count for each page **/
```

Comment:
This shouldn't reference `DataPageHeaderV2` or `Statistics`. It should just have a clear description, "Number of null values in the page".